src/Ombi.TheMovieDbApi/TheMovieDbApi.cs

```diff
@@ -99,65 +105,95 @@ public async Task<List<MovieSearchResult>> SimilarMovies(int movieId, string lan
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            request.AddQueryString("vote_count.gte", "250");
+            request.AddQueryString("sort_by", "vote_average.desc");
+            var settings = await Settings;
+            request.AddQueryString("include_adult", settings.ShowAdultMovies.ToString().ToLower());
+            request.AddQueryString("without_keywords", settings.ExcludedKeywordIds);
+
             AddRetry(request);
             var result = await Api.Request<TheMovieDbContainer<SearchResult>>(request);
             return Mapper.Map<List<MovieSearchResult>>(result.results);
         }
 
-        public async Task<List<MovieSearchResult>> Upcoming(string langageCode)
+        public async Task<List<MovieSearchResult>> Upcoming(string langCode)
         {
-            var request = new Request($"movie/upcoming", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            request.AddQueryString("with_release_type", "2|3");
+            var startDate = DateTime.Today.AddDays(7);
+            request.AddQueryString("release_date.gte", startDate.ToString("yyyy-MM-dd"));
+            request.AddQueryString("release_date.lte", startDate.AddDays(17).ToString("yyyy-MM-dd"));
+            var settings = await Settings;
+            request.AddQueryString("include_adult", settings.ShowAdultMovies.ToString().ToLower());
+            request.AddQueryString("without_keywords", settings.ExcludedKeywordIds);
+
             AddRetry(request);
             var result = await Api.Request<TheMovieDbContainer<SearchResult>>(request);
             return Mapper.Map<List<MovieSearchResult>>(result.results);
         }
 
-        public async Task<List<MovieSearchResult>> NowPlaying(string langageCode)
+        public async Task<List<MovieSearchResult>> NowPlaying(string langCode)
         {
-            var request = new Request($"movie/now_playing", BaseUri, HttpMethod.Get);
-            request.FullUri = request.FullUri.AddQueryParameter("api_key", ApiToken);
-            request.FullUri = request.FullUri.AddQueryParameter("language", langageCode);
+            var request = new Request($"discover/movie", BaseUri, HttpMethod.Get);
+            request.AddQueryString("api_key", ApiToken);
+            request.AddQueryString("language", langCode);
+            request.AddQueryString("with_release_type", "2|3");
+            var today = DateTime.Today;
+            request.AddQueryString("release_date.gte", today.AddDays(-42).ToString("yyyy-MM-dd"));
```

Comment:
Also why these dates too?

Reply:
@tidusjar Same reason as `movie/upcoming` for `movie/now_playing`. The endpoint returns an arbitrary date range without specifying in the docs what logic it is using.